src/Microsoft.Extensions.OptionsModel/IOptionsWatcher.cs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.OptionsModel
+{
+    public interface IOptionsWatcher<out TOptions>
```

Comment:
Are you in love with that name?  Not a huge fan so I was trying some others on for size... IOptionsMonitor also has the same semantic meaning as Watcher.  I view this class as the similar to FileMonitor/Watcher, as you get notified when your options change